File: src/main/java/com/ulbs/careerstartup/dto/JobCandidatesDTO.java
package com.ulbs.careerstartup.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotEmpty;
import lombok.*;

import jakarta.validation.constraints.NotNull;
import java.sql.Timestamp;
import java.util.UUID;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
@EqualsAndHashCode
public class JobCandidatesDTO {

    private UUID candidateId;

    private UUID jobId;

    private String applicationDate;

    @NotNull(message = "User is required")
    @NotEmpty(message = "User is required")
    @JsonProperty("user")
    private UserDTO userDTO;

    @NotNull(message = "PostedJob is required")
    @NotEmpty(message = "PostedJob is required")
    @JsonProperty("postedJob")
    private PostedJobDTO postedJobDTO;
}

File: src/app/services/models/job-candidates-dto.ts
/* tslint:disable */
/* eslint-disable */
import { PostedJobDto } from '../models/posted-job-dto';
import { UserDto } from '../models/user-dto';
export interface JobCandidatesDto {
  applicationDate?: string;
  candidateId?: string;
  jobId?: string;
  postedJob: PostedJobDto;
  user: UserDto;
}
